storage-proofs-porep/src/stacked/vanilla/params.rs

```diff
@@ -304,6 +308,77 @@ impl<Tree: MerkleTreeTrait, G: Hasher> Proof<Tree, G> {
 
         true
     }
+
+    #[allow(clippy::type_complexity)]
+    fn from_parts(
+        proof_d: MerkleProof<G, U2>,
+        col_proof: ColumnProof<
+            MerkleProof<Tree::Hasher, Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>,
+        >,
+        drg_col_proofs: Vec<
+            ColumnProof<
+                MerkleProof<Tree::Hasher, Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>,
+            >,
+        >,
+        exp_col_proofs: Vec<
+            ColumnProof<
+                MerkleProof<Tree::Hasher, Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>,
+            >,
+        >,
+        proof_r: MerkleProof<Tree::Hasher, Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>,
+    ) -> Self {
+        let challenge = col_proof.column.index as u64;
+        let num_layers = col_proof.column.rows.len();
+        let num_drg_parents = drg_col_proofs.len();
+        let num_exp_parents = exp_col_proofs.len();
+
+        let labeling_proofs: Vec<LabelingProof<Tree::Hasher>> = (0..num_layers)
+            .map(|layer_index| {
+                let has_exp_parents = (layer_index != 0) as usize;
+                let layer_parents = num_drg_parents + has_exp_parents * num_exp_parents;
```

Comment:
Casting a boolean into a usize is pretty C-like. I'd prefer having it in a more type-safe way. What about:

```rust
let layer_parents = if layer_index == 0 {
    num_drg_parents
} else {
    num_drg_parents + num_exp_parents
}

Reply:
Yup, sounds good.